### DISCUSSION TITLE
Sampler in Text vs. Sampler in System ?

### DISCUSSION THREAD

**Discussion by cgidesign-de** [vladmandic/sdnext#2929] [General]
In the Text tab I can select samplers from a list.
In the System tab I can set settings for the samplers.
So far, so good.
What I am unsure about is: how do they relate to each other?
E.g.
System setting is "sde-dpmsolver++"
Text dropdown offers "DPM++2M"
Does this mean that "DPM++2M" uses "sde" like set in settings?
Or, are the choices in the Text tab dropdown just presets that work independent from the System settings?
![Solver](https://github.com/vladmandic/automatic/assets/115958660/b625a72e-71da-4857-8ec4-4e9c2323fa1c)

**Comment by vladmandic** [comment]
options in txt2img ui and options in settings are combined when configuring a sampler, only difference why something is in txt2img ui is because its more commonly used.

note that not all options are applicable to all samplers - for example, dpm solver algorithm is only applicable to dpm++ 1s/2m samplers, which means they are always dpm++ and can be sde or non-sde. it has no meaning when applied to dpm sde.

yes, that area needs better documentation. i'd welcome contributions.

if you want to see which settings are applicable to which sampler, you can take a look at code: https://github.com/vladmandic/automatic/blob/8925cdc51ccc7432e25a388518e4d8bf9a4243a5/modules/sd_samplers_diffusers.py#L36-L52
